packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/EmailInvoiceDetails.tsx

```diff
@@ -72,17 +76,65 @@ const EmailInvoiceDetailsBase = ({
 }: EmailInvoiceDetailsProps) => {
   const { i18n } = useLingui();
   const { monite, api } = useMoniteContext();
-  const { control, handleSubmit, getValues, setValue, trigger } = useForm({
-    resolver: yupResolver(getEmailInvoiceDetailsSchema(i18n)),
-    defaultValues: {
-      subject: '',
-      body: '',
-      to: '',
-      // TODO: add support for multiple recipients, cc and bcc fields
-    },
-  });
+
+  const { control, handleSubmit, getValues, setValue, trigger, watch } =
+    useForm({
+      resolver: yupResolver(getEmailInvoiceDetailsSchema(i18n)),
+      defaultValues: {
+        subject: '',
+        body: '',
+        to: '',
+        // TODO: add support for multiple recipients, cc and bcc fields
+      },
+    });
+
+  const { data: me } = useMe();
+  const { data: receivable } = useReceivableById(invoiceId);
+  const { data: contacts } = useReceivableContacts(invoiceId);
+  const to = watch('to');
+  const body = watch('body');
+  const previousBody = useRef(body);
+  useEffect(() => {
+    if (me && to && contacts && body == previousBody.current) {
+      const contact = contacts.find(({ email }) => email == to);
+      if (contact) {
+        const newBody = t(i18n)`Hi ${contact.first_name},
+
+Please find the invoice attached as discussed.
+
+Kind Regards,
+${me.first_name}`;
```

Comment:
Can we please fix here identation problem, I see prettier can't format this easily